ghc/GHCi/UI.hs

```diff
@@ -1830,13 +1829,13 @@ markTargetsForRecompilation targets = do
             let newLinkable = linkable { linkableTime = beginningOfTime }
             in hmi { hm_linkable = Just newLinkable }
           _ -> hmi
-    findTargetModuleName _ (TargetModule name) = return $ Right name
+    findTargetModuleName _ (TargetModule name) = return name
     findTargetModuleName dflags (TargetFile fp _) = do
       buf <- hGetStringBuffer fp
       imports <- getImports dflags buf fp fp
-      return $ case imports of
-        (Right (_, _, GHC.L _ mname)) -> Right mname
-        Left err -> Left err
+      case imports of
+        (Right (_, _, GHC.L _ mname)) -> return mname
+        Left (bagToList -> err) -> throwOneError $ head err
```

Comment:
What's the effect of throwing only one error here? Do we miss reporting some errors?

Reply:
The error is packed by using `unitBug` in here https://www.stackage.org/haddock/lts-16.5/ghc-8.8.3/src/HeaderInfo.html#getImports

so it will always have and only have one error in this "case".

Do you suggest we should always assume there will be more, in case of future version of getImports does create more than one errors?

Reply:
Ah, I see. That's rather weird code but you're right.